src/pepr/operator/controllers/keycloak/client-sync.ts

```diff
@@ -61,10 +66,11 @@ export async function purgeSSOClients(pkg: UDSPackage, refs: string[] = []) {
   const currentClients = pkg.status?.ssoClients || [];
   const toRemove = currentClients.filter(client => !refs.includes(client));
   for (const ref of toRemove) {
-    const token = Store.getItem(ref);
+    const sanitizedRef = sanitizeResourceName(ref);
+    const token = Store.getItem(sanitizedRef);
```

Comment:
Do we need to account for changing the client key in the store?  For existing clients, the client id won't change but the key in the store likely will?

Reply:
I need to validate if there are any cases where the key could change. Essentially it would have to be something where `name` (client id) is different from the sanitized name, but valid as a pepr store key. I'll test a few things like length, different characters, and case sensitivity. I'm pretty confident that nothing in core/swf falls into that category but will check a few cases to test.

If we have to I think we could add a startup function that copies/migrates all sso keys from `name` -> `sanitizeResourceName(name)`.

Reply:
Encountered a few issues here - working on fixes and figuring out what is allowable in each place that we use the client ID. I think currently there are a few spots where things would conflict on an upgrade or even on switches to clientID if the new clientID overlaps with the old one once sanitized.

Reply:
Going to leave this PR in draft for now - after exploring this further I encountered several issues with how/where we were managing things that use the client id. I'm going to split off the Package retry and logging changes into a new PR/branch so that those can be independently reviewed + merged and give us more time to consider sane management of tokens + _weird_ client ids.